pkg/destroy/gcp/forwardingrule.go

```diff
@@ -16,15 +16,36 @@ const (
 	regionForwardingRuleResource = "regionalforwardingrule"
 )
 
+func (o *ClusterUninstaller) deleteForwardingRuleByName(ctx context.Context, resourceName, location string) error {
+	typeName := regionForwardingRuleResource
+	if location == string(global) {
+		typeName = globalForwardingRuleResource
+	}
+	items, err := o.listForwardingRulesWithFilter(ctx, typeName, "items(name,labels),nextPageToken", func(item *compute.ForwardingRule) bool {
+		return item.Name == resourceName && o.isOwnedResource(item.Labels)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list forwarding rules by name: %w", err)
+	}
+	for _, item := range items {
+		if err := o.deleteForwardingRule(ctx, item); err != nil {
+			return fmt.Errorf("failed to delete forwarding rule by name: %w", err)
+		}
+	}
+	return nil
+}
+
 func (o *ClusterUninstaller) listForwardingRules(ctx context.Context, typeName string) ([]cloudResource, error) {
-	return o.listForwardingRulesWithFilter(ctx, typeName, "items(name,region,loadBalancingScheme),nextPageToken", o.isClusterResource)
+	return o.listForwardingRulesWithFilter(ctx, typeName, "items(name,region,loadBalancingScheme,labels),nextPageToken", func(item *compute.ForwardingRule) bool {
+		return o.isClusterResource(item.Name) && o.isOwnedResource(item.Labels)
```

Comment:
There are other places (i.e. `listImages` in `pkg/destroy/gcp/image.go`) that check like:

 `o.isClusterResource(item.Name) && !o.isSharedResource(item.Labels)`.

Should we make it consistent?